SharkCage/SharedFunctionality/ValidateBinary.cpp

```diff
@@ -0,0 +1,226 @@
+#include "stdafx.h"
+#include "ValidateBinary.h"
+
+#include <tchar.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <windows.h>
+#include <Softpub.h>
+#include <wincrypt.h>
+#include <wintrust.h>
+#include <fstream>
+
+#pragma comment(lib, "wintrust")
+#pragma comment(lib, "bcrypt.lib")
+
+#define NT_SUCCESS(Status)          (((NTSTATUS)(Status)) >= 0)
+#define STATUS_UNSUCCESSFUL         ((NTSTATUS)0xC0000001L)
+
+bool ValidateBinary::ValidateCertificate(const std::wstring &app_path)
+{
+	WINTRUST_FILE_INFO FileData = { 0 };
+	FileData.cbStruct = sizeof(WINTRUST_FILE_INFO);
+	FileData.pcwszFilePath = app_path.c_str();
+	FileData.hFile = nullptr;
+	FileData.pgKnownSubject = nullptr;
+
+	WINTRUST_DATA WinTrustData = { 0 };
+	WinTrustData.cbStruct = sizeof(WinTrustData);
+	WinTrustData.pPolicyCallbackData = nullptr;
+	WinTrustData.pSIPClientData = nullptr;
+	WinTrustData.dwUIChoice = WTD_UI_NONE;
+	WinTrustData.fdwRevocationChecks = WTD_REVOKE_NONE;
+	WinTrustData.dwUnionChoice = WTD_CHOICE_FILE;
+	WinTrustData.dwStateAction = WTD_STATEACTION_VERIFY;
+	WinTrustData.hWVTStateData = nullptr;
+	WinTrustData.pwszURLReference = nullptr;
+	WinTrustData.dwUIContext = 0;
+	WinTrustData.pFile = &FileData;
+
+	GUID WVTPolicyGUID = WINTRUST_ACTION_GENERIC_VERIFY_V2;
+
+	LONG status = ::WinVerifyTrust(
+		nullptr,
+		&WVTPolicyGUID,
+		&WinTrustData);
+
+	WinTrustData.dwStateAction = WTD_STATEACTION_CLOSE;
```

Comment:
Because of that the `WTD_STATEACTION_VERIFY` was set according to the docu we had to close the action but because of your suggestion I changed it to `WTD_STATEACTION_IGNORE ` that makes this line superfluous.

Reply:
ah okay, but I think you would have to call the function again with this flag for it to have any effect. but if it's gone anyway... :)